app/uk.gov.hmrc.helptosavestridefrontend/connectors/HelpToSaveConnector.scala

```diff
@@ -186,11 +186,9 @@ class HelpToSaveConnectorImpl @Inject() (http:              HttpClient,
           Right(AccountAlreadyExists)
 
         case _ =>
-          logger.warn(s"createAccount returned a status: ${response.status} " +
-            s"with response body: ${maskNino(response.body)}", nSIUserInfo.nino)
+          logger.warn(s"createAccount returned a status: ${response.status}, ${nSIUserInfo.nino}")
```

Comment:
This logs nino, the original masks nino.

Reply:
maskNino was only masking the response.body, isn't it? We removed only that sections so we didn't add anything new, and as we decided today NINO is not PII in isolation

Reply:
Subtracting string interpolation, the original returns the following:

```scala
response.status // integer
maskNino(response.body)} // string with NINOs removed
```

And calls the [following extension method](https://github.com/hmrc/help-to-save-stride-frontend/blob/0a7ffb8253852f36767f4badce66052aeec1b652/app/uk.gov.hmrc.helptosavestridefrontend/util/Logging.scala#L38) on the logger object. This method accepts both a message and a NINO. [(type NINO = String)](https://github.com/hmrc/help-to-save-stride-frontend/blob/0a7ffb8253852f36767f4badce66052aeec1b652/app/uk.gov.hmrc.helptosavestridefrontend/util/package.scala#L34). Depending on the value of the config setting `nino-logging.enabled`, which is [set to false on all environments](https://catalogue.tax.service.gov.uk/service/help-to-save-stride-frontend/config#ws.timeout.request) nino may be enabled to be logged.

[The mask nino function](https://github.com/hmrc/help-to-save-stride-frontend/blob/main/app/uk.gov.hmrc.helptosavestridefrontend/util/package.scala#L38) by regex looks for things resembling ninos and replaces them with a string `"<NINO>"`. If this setting is enabled, the function appends `"with NINO [$nino]"` to log output.

Yes, it only removes NINOs from the response body, but response body is the only thing that contains any NINOs (arbitrary JSON with a bunch of user info which we definitely should not log).

Ninos are allowed to be logged, but for this particular service they were not logged in the first place.

We were asked to remove/reduce logs such that PII is not logged, not add more logs arbitrarily. I would expect this to be replaced with the following:

```scala
logger.warn(s"createAccount returned a status: ${response.status}", nino);
```

This will preserve the nino logging functionality as it was and will cease to log the entire response body which does contain PII.